File: snippets/messaging/purchase-phone-number/purchase-phone-number.js
const accountSid = 'ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX';
const authToken = 'your_auth_token';
const client = require('twilio')(accountSid, authToken);

// Find and then purchase a phone number
client
  .availablePhoneNumbers('US')
  .local.list({
    areaCode: '510',
  })
  .then(data => {
    const number = data[0];
    return client.incomingPhoneNumbers.create({
      phoneNumber: number.phoneNumber,
    });
  })
  .then(purchasedNumber => console.log(purchasedNumber.sid))
  .done();

File: snippets/messaging/purchase-phone-number/purchase-phone-number.cs
const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
const string authToken = "your_auth_token";
TwilioClient.Init(accountSid, authToken);

var localAvailableNumbers = LocalResource.Read("US", areaCode: 510);

// Purchase the first number on the list
var firstNumber = localAvailableNumbers.First();
var incomingPhoneNumber = IncomingPhoneNumberResource.Create(
    phoneNumber: firstNumber.PhoneNumber
);

Console.WriteLine(incomingPhoneNumber.Sid);
